openwisp_radius/api/views.py

```diff
@@ -84,13 +85,42 @@
 
 
 class FreeradiusApiAuthentication(BaseAuthentication):
+    def _get_ip_list(self, uuid):
+        if f'ip-{uuid}' in cache:
+            ip_list = cache.get(f'ip-{uuid}')
+        else:
+            try:
+                ip_list = OrganizationRadiusSettings.objects.get(
+                    organization__pk=uuid
+                ).freeradius_allowed_hosts_list
+            except OrganizationRadiusSettings.DoesNotExist:
+                ip_list = None
+            else:
+                cache.set(f'ip-{uuid}', ip_list)
+        return ip_list
+
+    def _check_client_ip_and_return(self, request, uuid):
+        client_ip, _is_routable = get_client_ip(request)
+        ip_list = self._get_ip_list(uuid)
+        if bool(
+            (ip_list and client_ip in ip_list)
+            or (not ip_list and client_ip in app_settings.FREERADIUS_ALLOWED_HOSTS)
+        ):
+            return (AnonymousUser(), uuid)
+        message = _(
+            f'Request rejected: Client IP address ({client_ip}) '
+            'is not allowed by the organization admin for freeradius API.'
```

Comment:
I would change this to:

`is not in the list of IP addresses allowed to consume the freeradius API`

@morrme any useful suggestion?